print(file=foo, flush=True) should require foo to have a flush method

https://github.com/python/typeshed/blob/aa7e277adb9049e24ea3434fc9848defbfa87673/stdlib/builtins.pyi#L1278-L1280

but:

```
>>> class Foo:
...     def write(self, s):
...             pass
... 
>>> print("hello", file=Foo())
>>> print("hello", file=Foo(), flush=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Foo' object has no attribute 'flush'
>>> 
```

This seems to work:

```
from _typeshed import SupportsWrite
from typing import Protocol, TypeVar, overload
from typing_extensions import Literal

_T = TypeVar('_T', contravariant=True)

class SupportsFlush(SupportsWrite[_T], Protocol[_T]):
    def flush(self) -> None: ...

@overload
def print(
    *values: object, sep: str | None = ..., end: str | None = ..., file: SupportsWrite[str] | None = ..., flush: Literal[False] = ...
) -> None: ...

@overload
def print(
    *values: object, sep: str | None = ..., end: str | None = ..., file: SupportsFlush[str] | None = ..., flush: bool = ...
) -> None: ...

class Foo:
    def write(self, s: str) -> None: ...
    
class Bar:
    def write(self, s: str) -> None: ...
    def flush(self) -> None: ...

print('blah', file=Foo())
print('blah', file=Foo(), flush=False)
print('blah', file=Foo(), flush=True)  # error: No overload variant of "print" matches argument types "str", "Foo", "bool"  [call-overload]
```

Would it be better to put "SupportsFlush" in `_typeshed`, or have it as a private protocol in `builtins`?

--- a/stdlib/builtins.pyi
+++ b/stdlib/builtins.pyi
@@ -1275,8 +1275,21 @@ def open(
     opener: _Opener | None = ...,
 ) -> IO[Any]: ...
 def ord(__c: str | bytes) -> int: ...
+
+class _SupportsWriteAndFlush(SupportsWrite[_T_contra], Protocol[_T_contra]):
+    def flush(self) -> None: ...
+
+@overload
 def print(
-    *values: object, sep: str | None = ..., end: str | None = ..., file: SupportsWrite[str] | None = ..., flush: bool = ...
+    *values: object,
+    sep: str | None = ...,
+    end: str | None = ...,
+    file: SupportsWrite[str] | None = ...,
+    flush: Literal[False] = ...,
+) -> None: ...
+@overload
+def print(
+    *values: object, sep: str | None = ..., end: str | None = ..., file: _SupportsWriteAndFlush[str] | None = ..., flush: bool
 ) -> None: ...
 
 _E = TypeVar("_E", contravariant=True)

